packages/fxa-settings/src/pages/InlineRecoveryKeySetup/index.tsx

```diff
@@ -21,6 +22,25 @@ export const InlineRecoveryKeySetup = ({
   createRecoveryKeyHandler: () => Promise<void>;
   currentStep: number;
 } & RouteComponentProps) => {
+  const doLaterHandler = () => {
+    localStorage.setItem(Constants.DISABLE_PROMO_ACCOUNT_RECOVERY_KEY, 'true');
+    // We do a hard navigate because this page is still in the content server, this
+    // also keeps all query params so that correct metrics are emitted.
+    hardNavigate('/connect_another_device', {}, true);
+  };
+
+  // We don't show this promo if the user has dismissed it or dismissed it from
+  // settings
+  if (
```

Comment:
Should this check happen at the previous step (i.e., in the `getSyncNavigate` function)?